core/src/main/java/org/apache/accumulo/core/client/admin/compaction/CompactionConfigurer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.admin.compaction;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.accumulo.core.client.PluginEnvironment;
+import org.apache.accumulo.core.data.TableId;
+
+/**
+ * Enables dynamically overriding of configuration used to create the output file for a compaction.
+ *
+ * @since 2.1.0
+ */
+public interface CompactionConfigurer {
```

Comment:
What about calling this interface ```ExternalCompaction```?  Putting external in the name would at least provide a high level idea for the motivation behind this set of plugin components.

Reply:
Unless this is one of the intermediate interfaces, then that name wouldn't be appropriate.  It is difficult to distinguish the hierarchy just looking at the PR.

Reply:
> It is difficult to distinguish the hierarchy just looking at the PR.

Did you see the diagram in the javadoc I linked to?

Reply:
I had not, but I found it, thanks.
https://github.com/apache/accumulo/blob/8dd667b11e3b92c097ce78d596875b35e375a126/core/src/main/java/org/apache/accumulo/core/spi/compaction/doc-files/compaction-spi-design.png

Reply:
I also posted a copy of the rendered javadocs.

https://keith-turner.github.io/apidocs-accumulo-564/org/apache/accumulo/core/spi/compaction/package-summary.html